### PR TITLE
[LP-3] Redesign theory details page

### PR DESCRIPTION
**Pull request by IvanYarochkin**

### CODE REVIEW

File: src/main/webapp/app/entities/theory/theory.module.ts [hunk 1]
```diff
@@ -2,17 +2,17 @@ import { NgModule } from '@angular/core';
 import { RouterModule } from '@angular/router';
 
 import { LearnportalSharedModule } from 'app/shared/shared.module';
-import { TheoryComponent } from './theory.component';
 import { TheoryDetailComponent } from './theory-detail.component';
 import { TheoryUpdateComponent } from './theory-update.component';
 import { TheoryDeletePopupComponent, TheoryDeleteDialogComponent } from './theory-delete-dialog.component';
 import { theoryRoute, theoryPopupRoute } from './theory.route';
+import { SafeHtmlPipe } from 'app/pipes/safe-html.pipe';
```
Comment (xxMikhailx) [inline src/main/webapp/app/entities/theory/theory.module.ts]: Extract it to SharedModule

Comment (IvanYarochkin) [reply src/main/webapp/app/entities/theory/theory.module.ts]: Done.

File: src/main/webapp/app/entities/category/category.module.ts [hunk 2]
```diff
@@ -12,13 +11,7 @@ const ENTITY_STATES = [...categoryRoute, ...categoryPopupRoute];
...
 @NgModule({
   imports: [LearnportalSharedModule, RouterModule.forChild(ENTITY_STATES)],
-  declarations: [
-    CategoryComponent,
-    CategoryDetailComponent,
-    CategoryUpdateComponent,
-    CategoryDeleteDialogComponent,
-    CategoryDeletePopupComponent
-  ],
+  declarations: [CategoryDetailComponent, CategoryUpdateComponent, CategoryDeleteDialogComponent, CategoryDeletePopupComponent],
```
Comment (xxMikhailx) [inline src/main/webapp/app/entities/category/category.module.ts:14]: Format it like one component per line

File: src/main/webapp/app/entities/category/category.route.ts [hunk 3]
```diff
@@ -73,6 +78,51 @@ export const categoryRoute: Routes = [
       pageTitle: 'learnportalApp.category.home.title'
     },
     canActivate: [UserRouteAccessService]
+  },
+  {
+    path: ':categoryId/theory',
```
Comment (xxMikhailx) [inline src/main/webapp/app/entities/category/category.route.ts]: Change all `categoryId` to simple `id` as we use it in category route

File: src/main/webapp/app/entities/deck/deck.module.ts [hunk 4]
```diff
@@ -12,7 +11,7 @@ const ENTITY_STATES = [...deckRoute, ...deckPopupRoute];
 
 @NgModule({
   imports: [LearnportalSharedModule, RouterModule.forChild(ENTITY_STATES)],
-  declarations: [DeckComponent, DeckDetailComponent, DeckUpdateComponent, DeckDeleteDialogComponent, DeckDeletePopupComponent],
+  declarations: [DeckDetailComponent, DeckUpdateComponent, DeckDeleteDialogComponent, DeckDeletePopupComponent],
```
Comment (xxMikhailx) [inline src/main/webapp/app/entities/deck/deck.module.ts:14]: Format it like one component per line

File: src/main/webapp/app/entities/formula/formula.module.ts [hunk 5]
```diff
@@ -12,13 +11,7 @@ const ENTITY_STATES = [...formulaRoute, ...formulaPopupRoute];
...
 @NgModule({
   imports: [LearnportalSharedModule, RouterModule.forChild(ENTITY_STATES)],
-  declarations: [
-    FormulaComponent,
-    FormulaDetailComponent,
-    FormulaUpdateComponent,
-    FormulaDeleteDialogComponent,
-    FormulaDeletePopupComponent
-  ],
+  declarations: [FormulaDetailComponent, FormulaUpdateComponent, FormulaDeleteDialogComponent, FormulaDeletePopupComponent],
```
Comment (xxMikhailx) [inline src/main/webapp/app/entities/formula/formula.module.ts:14]: Format it like one component per line

File: src/main/webapp/app/entities/quiz/quiz.module.ts [hunk 6]
```diff
@@ -12,7 +11,7 @@ const ENTITY_STATES = [...quizRoute, ...quizPopupRoute];
 
 @NgModule({
   imports: [LearnportalSharedModule, RouterModule.forChild(ENTITY_STATES)],
-  declarations: [QuizComponent, QuizDetailComponent, QuizUpdateComponent, QuizDeleteDialogComponent, QuizDeletePopupComponent],
+  declarations: [QuizDetailComponent, QuizUpdateComponent, QuizDeleteDialogComponent, QuizDeletePopupComponent],
```
Comment (xxMikhailx) [inline src/main/webapp/app/entities/quiz/quiz.module.ts:14]: Format it like one component per line

File: src/main/webapp/app/entities/task/task.module.ts [hunk 7]
```diff
@@ -12,7 +11,7 @@ const ENTITY_STATES = [...taskRoute, ...taskPopupRoute];
 
 @NgModule({
   imports: [LearnportalSharedModule, RouterModule.forChild(ENTITY_STATES)],
-  declarations: [TaskComponent, TaskDetailComponent, TaskUpdateComponent, TaskDeleteDialogComponent, TaskDeletePopupComponent],
+  declarations: [TaskDetailComponent, TaskUpdateComponent, TaskDeleteDialogComponent, TaskDeletePopupComponent],
```
Comment (xxMikhailx) [inline src/main/webapp/app/entities/task/task.module.ts:14]: Format it like one component per line

File: src/main/webapp/app/entities/theory/theory.module.ts [hunk 8]
```diff
@@ -12,7 +11,7 @@ const ENTITY_STATES = [...theoryRoute, ...theoryPopupRoute];
 
 @NgModule({
   imports: [LearnportalSharedModule, RouterModule.forChild(ENTITY_STATES)],
-  declarations: [TheoryComponent, TheoryDetailComponent, TheoryUpdateComponent, TheoryDeleteDialogComponent, TheoryDeletePopupComponent],
+  declarations: [TheoryDetailComponent, TheoryUpdateComponent, TheoryDeleteDialogComponent, TheoryDeletePopupComponent],
```
Comment (xxMikhailx) [inline src/main/webapp/app/entities/theory/theory.module.ts:14]: Format it like one component per line

**Review by xxMikhailx — APPROVED** [review]

File: src/main/webapp/app/entities/category/category.route.ts (hunk 3 above)
Comment (IvanYarochkin) [reply src/main/webapp/app/entities/category/category.route.ts]: Done.

File: src/main/webapp/app/entities/category/category.module.ts (hunk 2 above)
Comment (IvanYarochkin) [reply src/main/webapp/app/entities/category/category.module.ts:14]: Don't change as discussed.

File: src/main/webapp/app/entities/deck/deck.module.ts (hunk 4 above)
Comment (IvanYarochkin) [reply src/main/webapp/app/entities/deck/deck.module.ts:14]: Don't change as discussed.

File: src/main/webapp/app/entities/formula/formula.module.ts (hunk 5 above)
Comment (IvanYarochkin) [reply src/main/webapp/app/entities/formula/formula.module.ts:14]: Don't change as discussed.

File: src/main/webapp/app/entities/quiz/quiz.module.ts (hunk 6 above)
Comment (IvanYarochkin) [reply src/main/webapp/app/entities/quiz/quiz.module.ts:14]: Don't change as discussed.

File: src/main/webapp/app/entities/task/task.module.ts (hunk 7 above)
Comment (IvanYarochkin) [reply src/main/webapp/app/entities/task/task.module.ts:14]: Don't change as discussed.

File: src/main/webapp/app/entities/theory/theory.module.ts (hunk 8 above)
Comment (IvanYarochkin) [reply src/main/webapp/app/entities/theory/theory.module.ts:14]: Don't change as discussed.

**Pull request merged**